### PR TITLE
Namespace cleanup before 1.7

### PR DESCRIPTION
**Pull request by albanD**
Things that were not in 1.6
- for torch.: TYPE_CHECKING, textwrap, handle_torch_function, has_torch_function, DisableTorchFunction
- for torch.functional.: Any, Sequence, TYPE_CHECKING, Union
- for torch.nn.functional: Tuple

Things that were in 1.6 but removed now for consistency
- for torch.: Set, Type
- for torch.functional: Tuple, Optional
- for torch.nn.functional: List, Optional

### CODE REVIEW

**Comment by albanD** [comment]
Note that the lint error is due to the temporary removal of the "slowTest" so the test will pass when the PR is ready for merge.

**Review by rgommers — APPROVED** [review]
Thanks @albanD. Both the typing and `__torch_function__` related changes LGTM.

File: test/benchmark_utils/test_benchmark_utils.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ class MockCudaTimer(benchmark_utils.Timer):
               20 measurements, 1000 runs per measurement, 16 threads"""
         )
 
-    @slowTest
+    # Just to make sure this test runs in PR ci will be added back before merging
```
Comment (gchanan) [inline test/benchmark_utils/test_benchmark_utils.py:439]: can't you put this on the ci-all branch?

File: torch/__init__.py [hunk 2]
```diff
@@ -170,6 +170,7 @@ def _load_global_deps():
             import torch._dl as _dl_flags  # type: ignore
     old_flags = sys.getdlopenflags()
     sys.setdlopenflags(_dl_flags.RTLD_GLOBAL | _dl_flags.RTLD_LAZY)
+    # Note that this import ignores all the functions whose name that starts with "_"
```
Comment (gchanan) [inline torch/__init__.py:173]: nit: link to where this is documented in the python docs?

File: torch/__init__.py [hunk 3]
```diff
@@ -187,11 +188,13 @@ def _load_global_deps():
     # See Note [Global dependencies]
     if USE_GLOBAL_DEPS:
         _load_global_deps()
+
+    # Note that this import ignores all the functions whose name that starts with _
```
Comment (gchanan) [inline torch/__init__.py:192]: same here.

File: torch/utils/benchmark/utils/valgrind_wrapper/timer_interface.py [hunk 4]
```diff
@@ -743,13 +743,13 @@ def check_result(completed_process):
             # =============================================================================
             # == User code block ==========================================================
             # =============================================================================
-            callgrind_bindings.valgrind_toggle()
+            callgrind_bindings._valgrind_toggle()
```
Comment (gchanan) [inline torch/utils/benchmark/utils/valgrind_wrapper/timer_interface.py:746]: can you split the valgrind change into it's own PR?

File: torch/nn/functional.py [hunk 5]
```diff
@@ -9,7 +9,8 @@
 from .modules.utils import _single, _pair, _triple, _list_with_default
 from . import grad  # noqa: F401
 from torch import _VF
-from .._jit_internal import boolean_dispatch, List, Optional, _overload, Tuple
+from .._jit_internal import boolean_dispatch, _overload
+from typing import List, Optional, Tuple
```
[the remark this answers is not in the document]
Comment (albanD) [reply torch/nn/functional.py:13]: Yes the other file just import them from typing and does no change on them.
I updated here to make this clearer that they are the plain version.

File: torch/utils/benchmark/utils/valgrind_wrapper/timer_interface.py (hunk 4 above)
Comment (albanD) [reply torch/utils/benchmark/utils/valgrind_wrapper/timer_interface.py:746]: Sure

File: test/benchmark_utils/test_benchmark_utils.py (hunk 1 above)
Comment (albanD) [reply test/benchmark_utils/test_benchmark_utils.py:439]: Will do that for the valgrind PR

File: torch/__init__.py (hunk 2 above)
Comment (albanD) [reply torch/__init__.py:173]: It is not very well documented. But mentioned here: https://docs.python.org/3/tutorial/modules.html#importing-from-a-package
Will add a link to this in the comment.

**Comment by albanD** [comment]
Typing changes with `_jit_internal.py` seem to have unexpected behavior.
After chatting with Greg, I'll remove these for now to avoid any risk for 1.7

**Comment by albanD** [comment]
Cancelling for now for more specific and safer PRs.
